lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -111,8 +115,89 @@ struct CallScheduleable {
 };
 
 /// A variant of types representing scheduleable operations.
-using Scheduleable = std::variant<calyx::GroupOp, WhileScheduleable,
-                                  ForScheduleable, CallScheduleable>;
+using Scheduleable =
+    std::variant<calyx::GroupOp, IfScheduleable, WhileScheduleable,
+                 ForScheduleable, CallScheduleable>;
+
+class IfLoweringStateInterface {
+public:
+  void addThenYieldRegs(scf::IfOp op, calyx::RegisterOp reg, unsigned idx) {
+    assert(
+        thenYieldRegs[op.getOperation()].count(idx) == 0 &&
+        "A register was already registered for the given then yield args.\n");
+    assert(idx < op->getNumOperands());
+    thenYieldRegs[op.getOperation()][idx] = reg;
+  }
+
+  const DenseMap<unsigned, calyx::RegisterOp> &getThenYieldRegs(scf::IfOp op) {
+    return thenYieldRegs[op.getOperation()];
+  }
+
+  calyx::RegisterOp getThenYieldRegs(scf::IfOp op, unsigned idx) {
+    auto regs = getThenYieldRegs(op);
+    auto it = regs.find(idx);
+    assert(it != regs.end() &&
+           "No then yield regs set for the provided index!");
+    return it->second;
+  }
+
+  void addElseYieldRegs(scf::IfOp op, calyx::RegisterOp reg, unsigned idx) {
+    assert(
+        elseYieldRegs[op.getOperation()].count(idx) == 0 &&
+        "A register was already registered for the given else yield args.\n");
+    assert(idx < op->getNumOperands());
+    elseYieldRegs[op.getOperation()][idx] = reg;
+  }
+
+  const DenseMap<unsigned, calyx::RegisterOp> &getElseYieldRegs(scf::IfOp op) {
+    return elseYieldRegs[op.getOperation()];
+  }
+
+  calyx::RegisterOp getElseYieldRegs(scf::IfOp op, unsigned idx) {
+    auto regs = getElseYieldRegs(op);
+    auto it = regs.find(idx);
+    assert(it != regs.end() &&
```

Comment:
nit: Again, we aren't worried about the assertion path and generally lean towards shorter code.